### PR TITLE
adding cross-validation folds for the purpose of testing supervised m…

### PR DESCRIPTION
**Pull request by jwist**
…ethods

The idea is to add fixed cross-validation folds for testing statistics in supervised methods. It is important to be able to compare the results obtained using the same "random" selection during cross-validation. 

Here we added two different sets for 7-folds and two different sets for 5-folds.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+declare module 'ml-iris-dataset' {
```
Comment (targos) [inline index.d.ts]:
```suggestion
declare module 'ml-dataset-iris' {
```

File: src/iris.js [hunk 2]
```diff
@@ -1,19 +1,75 @@
...
+exports.getCrossValidationSets = function(k, options) {
+    const {idx = 0,
+    by = 'folds'} = options;
+    var res;
+    k == 7 ? res = cvSetsK7[idx] : [];
+    k == 5 ? res = cvSetsK5[idx] : [];
+
+switch(by) {
+    case 'folds': {
+      res.map((x) => x.map((y) => y - 1));
```
Comment (targos) [inline src/iris.js]: this has no effect. [].map() returns a new array

File: src/iris.js [hunk 3]
```diff
@@ -1,19 +1,75 @@
...
+    }
+    break;
+    case 'trainTest': {
+      res.map((x) => x.map((y) => y - 1));
```
Comment (targos) [inline src/iris.js]: same here

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  export function getDataset()
+  
+  export function getNumbers()
+
+  export function getClasses()
+
+  export function getDistinctClasses()
+
+  export function getCrossValidationSets(k: number, options?: cvOptions): array;
+}
```
Comment (targos) [inline index.d.ts:49]: This should be a bit better:

```js
declare module 'ml-dataset-iris' {
  export interface cvOptions {
    idx?: number;
    by?: 'folds' | 'trainTest';
  }

  export function getDataset(): Array<Array<number | string>>;

  export function getNumbers(): Array<Array<number>>;

  export function getClasses(): Array<string>;

  export function getDistinctClasses(): Array<string>;

  export function getCrossValidationSets(
    k: number,
    options?: cvOptions,
  ): Array<any>;
}
```

File: src/iris.js (hunk 2 above)
Comment (jwist) [reply src/iris.js]: yes, indeed it made crash my tests in further packages

File: src/iris.js (hunk 3 above)
Comment (jwist) [reply src/iris.js]: a new commit corrects this

File: index.d.ts (hunk 4 above)
Comment (jwist) [reply index.d.ts:49]: ok, got it! I've done the corrections.

**Pull request merged**